src/components/screens/home/editor-screen.tsx

```diff
@@ -280,42 +253,265 @@ class EditorScreen extends React.Component<Props, State> {
+        <ListItem button>
+          <ListItemIcon>
+            <TagFacesIcon/>
+          </ListItemIcon>
+          <ListItemText>
+            { intent.name }
+          </ListItemText>
+          </ListItem>
+          ))
+          }
+      </List>
+    )
+  }
+
+  /**
+   * Fetches knots list for the story
+   */
+  private fetchData = async() => {
+    console.log("fetching test data")
+    const { accessToken } = this.props;
+
+    if (!accessToken) {
+      return;
+    }
+
+    const storiesApi = Api.getStoriesApi(accessToken)
+    const knotsApi = Api.getKnotsApi(accessToken)
+    const intentsApi = Api.getIntentsApi(accessToken)
+
+    const [ stories ] = await Promise.all<Story[]>([
+      storiesApi.listStories()
+    ]);  
+    let mainStoryId: string = stories[0].id as string;
+    const foundStory = await storiesApi.findStory({storyId: mainStoryId})
+    if (foundStory) {
+      let foundStoryId: string = foundStory.id as string;
+      const [ knotList ] = await Promise.all<Knot[]>([
+        knotsApi.listKnots({
+          storyId: foundStoryId
+        })
+      ]);  
+
+      const [ intentList ] = await Promise.all<Intent[]>([
+        intentsApi.listIntents({
+          storyId: foundStoryId
+        })
+      ]);  
+
+      this.setState({
+        currentStory: foundStory,
+        storyKnots : knotList,
+        storyIntents: intentList
+      });
+    }
+  }
+
+
+    /**
+   * Component did mount life cycle handler
+   */
+  public componentDidMount = () => {
+    //this.clearTestData()
+    //this.createTestData()
+    this.fetchData();
+  }
+
+  /**
+   * Clears all test data
+   */
+  private clearTestData = async() => {
+    console.log("cleaning test data")
+    const { accessToken } = this.props;
+
+    if (!accessToken) {
+      return;
+    }
+
+    const storiesApi = Api.getStoriesApi(accessToken)
+    const knotsApi = Api.getKnotsApi(accessToken)
+    const intentsApi = Api.getIntentsApi(accessToken)
+    
+    const [ allStories ] = await Promise.all<Story[]>([
+      storiesApi.listStories()
+    ]);
+
+    for (let story of allStories) {
+      let storyId: string = story.id as string;
+      //remove all knots
+      const [ allKnots ] = await Promise.all<Knot[]>([
+        knotsApi.listKnots({
+            storyId: storyId
+          })
+      ]);
+      for (let knot of allKnots) {
+        let knotId: string = knot.id as string;
+        knotsApi.deleteKnot(
+          {
+            storyId: storyId,
+            knotId: knotId
+          }
+        )
+      }
+
+      //remove all intents
+      const [ allIntents ] = await Promise.all<Intent[]>([
+        intentsApi.listIntents({
+            storyId: storyId
+          })
+      ]);
+      for (let intent of allIntents) {
+        let intentId: string = intent.id as string;
+        intentsApi.deleteIntent(
+          {
+            storyId: storyId,
+            intentId: intentId
+          }
+        )
+      }
+    }
+  }
```

Comment:
```suggestion
    const storiesApi = Api.getStoriesApi(accessToken);
    const knotsApi = Api.getKnotsApi(accessToken);
    const intentsApi = Api.getIntentsApi(accessToken);
    
    const allStories = await storiesApi.listStories();

    allStories.forEach(story => {
      const storyId: string = story.id || "";
      
      const [ allKnots, allIntents ] = await Promise.all([
        knotsApi.listKnots({ storyId: storyId }),
        intentsApi.listIntents({ storyId: storyId })
      ]);
      
      //remove all knots
      allKnots.forEach(knot => {
        const knotId = knot.id || "";
        knotsApi.deleteKnot(
          {
            storyId: storyId,
            knotId: knotId
          }
        );
      });
      
      //remove all intents
      allIntents.forEach(intent => {
        const intentId: string = intent.id as string;
        intentsApi.deleteIntent(
          {
            storyId: storyId,
            intentId: intentId
          }
        );
      });
    });
  }
```

Reply:
Remember semicolons. Arrays have these high level functions that can be used for looping. I have been told to always use const unless I plan to change the value of the variable at some point.